go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -212,6 +215,87 @@ func (p DoltDatabaseProvider) CreateDatabase(ctx *sql.Context, name string) erro
 	return dsess.AddDB(ctx, dbstate)
 }
 
+// CloneDatabaseFromRemote implements DoltDatabaseProvider interface
+func (p DoltDatabaseProvider) CloneDatabaseFromRemote(ctx *sql.Context, dbName, branch, remoteName, remoteUrl string, remoteParams map[string]string) error {
+	p.mu.Lock()
```

Comment:
This is potentially a loooong time to be holding a lock! I think this is okay for v1, since `dolt_clone()` will not be frequently used, but it would be good to follow up and shrink down the code that executes with this lock.